sdk/appcenter/src/main/java/com/microsoft/appcenter/channel/DefaultChannel.java

```diff
@@ -764,54 +765,75 @@ public synchronized void enqueue(@NonNull Log log, @NonNull final String groupNa
+        if (groupState.mBatchTimeInterval > MINIMUM_TRANSMISSION_INTERVAL) {
+            return resolveCustomTriggerInterval(groupState);
+        } else {
+            return resolveDefaultTriggerInterval(groupState);
+        }
+    }
+
+    @WorkerThread
+    private Long resolveCustomTriggerInterval(@NonNull GroupState groupState) {
+        long now = System.currentTimeMillis();
+        long startTimer = SharedPreferencesManager.getLong(START_TIMER_PREFIX + groupState.mName);
+        if (groupState.mPendingLogCount > 0) {
 
             /* The timer isn't started or has invalid value (start time in the future), so start it and store the current time. */
-            if (pendingLogCount > 0 && (startTimer == 0 || startTimer > now)) {
+            if (startTimer == 0 || startTimer > now) {
                 SharedPreferencesManager.putLong(START_TIMER_PREFIX + groupState.mName, now);
                 AppCenterLog.debug(LOG_TAG, "The timer value for " + groupState.mName + " has been saved.");
+                return groupState.mBatchTimeInterval;
             }
 
-            /* If the interval is over. */
-            else if (startTimer + batchTimeInterval < now) {
-
-                /* Send all logs without any additional timers. */
-                if (pendingLogCount > 0) {
-                    triggerIngestion(groupState);
-                } else {
-
-                    /* Remove the time only when there are no logs left. */
-                    SharedPreferencesManager.remove(START_TIMER_PREFIX + groupState.mName);
-                    AppCenterLog.debug(LOG_TAG, "The timer for " + groupState.mName + " channel finished.");
-                }
-                return;
-            }
+            /* Wait for the rest of the interval. */
+            return Math.max(groupState.mBatchTimeInterval - (now - startTimer), 0);
+        } else {
 
-            /* We still have to wait for the rest of the interval. */
-            else {
-                batchTimeInterval -= now - startTimer;
+            /* If the interval is over. */
+            if (startTimer + groupState.mBatchTimeInterval < now) {
+                SharedPreferencesManager.remove(START_TIMER_PREFIX + groupState.mName);
+                AppCenterLog.debug(LOG_TAG, "The timer for " + groupState.mName + " channel finished.");
             }
-        } else if (pendingLogCount >= groupState.mMaxLogsPerBatch) {
-            triggerIngestion(groupState);
-            return;
+            return null;
         }
+    }
 
-        /* Postpone triggering ingestion. */
-        if (pendingLogCount > 0 && !groupState.mScheduled) {
-            groupState.mScheduled = true;
-            mAppCenterHandler.postDelayed(groupState.mRunnable, batchTimeInterval);
+    private Long resolveDefaultTriggerInterval(@NonNull GroupState groupState) {
+        if (groupState.mPendingLogCount >= groupState.mMaxLogsPerBatch) {
+            return 0L;
```

Comment:
Can you elaborate?

Reply:
@annakocheshkova This piece isn't changed. It's like: if there are enough logs for "full" batch then send it immediately.